### PR TITLE
Raise NotImplementedError for datetime strings with UTC offset

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
Avoids e.g. `DatetimeIndex(["2022-07-22 00:00:00+02:00"])` from dropping the `+02:00` since timezones are not supported

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/core/column/column.py [hunk 1]
```diff
@@ -2533,6 +2533,14 @@ def _construct_array(
         if inferred_dtype == "interval":
             # Only way to construct an Interval column.
             return pd.array(arbitrary)
+        elif inferred_dtype == "string" and getattr(dtype, "kind", "") == "M":
```
Comment (bdice) [inline python/cudf/cudf/core/column/column.py:2536]: Is None a better default?
```suggestion
        elif inferred_dtype == "string" and getattr(dtype, "kind", None) == "M":
```

File: python/cudf/cudf/tests/test_datetime.py [hunk 2]
```diff
@@ -2150,6 +2150,11 @@ def test_daterange_pandas_compatibility():
     assert_eq(expected, actual)
 
 
+def test_strings_with_utcoffset_not_implemented():
```
Comment (bdice) [inline python/cudf/cudf/tests/test_datetime.py:2153]:
```suggestion
def test_strings_with_utc_offset_not_implemented():
```

File: python/cudf/cudf/core/column/column.py (hunk 1 above)
Comment (mroeschke) [reply python/cudf/cudf/core/column/column.py:2536]: Sure thing. Added

**Review by bdice — APPROVED** [review]

**Comment by mroeschke** [comment]
Sorry for the noise here. I got in git-trouble

**Pull request closed without merging**